docs/deployment.rst

```diff
@@ -56,6 +56,27 @@ whichever service you plan to use) and then running:
 Users can then clone this repository, and point their local `knowledge_repo`
 script at it using :code:`--repo <path_of_cloned_repository>`.
 
+If the git repo is hosted on a remote service like Github, then the knowledge
+repo server will periodically fetch updates, while running as a server. If
+the remote service requires authentication (e.g. a private repository on Github),
+then these requests will fail.
+
+Taking Github as an example, it's possible to use SSH to authenticate in order
+to allow the knowledge repo server access. To set this up:
```

Comment:
Github -> GitHub?

I would recommend that people use deploy keys (which are repository specific) rather than a personal ssh key, and can be granted read-only access. (See: https://developer.github.com/v3/guides/managing-deploy-keys/#deploy-keys).

It's also possible to bypass some of this configuration by preparing the ssh-agent in the environment to be preloaded with a specific key, for example: `ssh-add /decrypted_secrets/id_rsa`, and adding the host to the list of known hosts: `ssh-keyscan -t rsa -p <ssh port> <host> >> ~/.ssh/known_hosts`.

Thoughts?

Reply:
Oh that's good to know. Let me try to set that up internally, and then update the guide accordingly.

Reply:
Yep @matthewwardrop that's a much better solution and works. I learned something. I updated the guidance accordingly with your other edits.